core/arch/arm/tee/drv_mgt.c

```diff
@@ -0,0 +1,263 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2018 NXP
+ *
+ * File    drv_mgt.c
+ *
+ * Brief   Global management of drivers registered with REGISTER_DRIVER
+ *         macro and using the struct driver format (driver data and
+ *         operations).
+ *
+ * The concept of driver object is introduced in OP-TEE through the
+ * `struct driver` available in the `core/include/drivers/driver.h`
+ * include file.
+ *
+ * This object has been designed to allow power management at driver level,
+ * hence to prepare the driver to enter in power transition.
+ * This mechanism is useful for driver:
+ *   - using DMA (Direct Memory Access) transfers to a memory that will
+ *   be shutdown
+ *   - losing configuration because clocks and/or power domain are shutdown
+ *
+ * Exported APIs to be called in the power management framework (e.g. PSCI)
+ *
+ *  - Switch in a power mode
+ * TEE_Result drivers_pm_enter(enum drv_pwrmode mode, bool wait)
```

Comment:
Should TEE_Result be used as return value?  I am not very sure. @jforissier @etienne-lms @jenswi-linaro

Reply:
`TEE_Result` is OK, that's what we normally use as internal error code.